broker-node/app/Clients/BrokerNode.php

```diff
@@ -380,25 +375,24 @@ public static function buildTransactionData($chunks)
         return $request;
     }
 
-    public static function getTransactionsToApprove(&$request)
+    private static function getTransactionsToApprove(&$request)
     {
         self::initIri();
 
         $command = new \stdClass();
         $command->command = "getTransactionsToApprove";
         $command->depth = IriData::$depthToSearchForTxs;
 
-        BrokerNode::$iriRequestInProgress = true;
 
         $result = self::$IriWrapper->makeRequest($command);
 
-        BrokerNode::$iriRequestInProgress = false;
 
         if (!is_null($result) && property_exists($result, 'branchTransaction')) {
             //switching trunk and branch
             //do we do this randomly or every time?
+            $tmp_trunk = $request->trunkTransaction;
             $request->trunkTransaction = $result->branchTransaction;
-            $request->branchTransaction = $result->trunkTransaction;
+            $request->branchTransaction = $tmp_trunk;
```

Comment:
Unless this is some assigning by reference, I'm not sure this will work?  $request->trunkTransaction is undefined at the time tmp_trunk is assigned.  $request->trunkTransaction and $request->branchTransaction are both empty, and get assigned to the values of $result->trunkTransaction and $result->branchTransaction

Reply:
Reverted, I misread this as as swapping instead of assigning to a $request object.